src/bluehawk/parser/lexer/makeLexer.ts

```diff
@@ -8,23 +8,25 @@ import {
   Text,
   AttributeListStart,
   Identifier,
+  BlockCommentEnd,
 } from "./tokens";
 import { tokenCategoryFilter } from "./tokenCategoryFilter";
 
-// After a command start tag, there may be an attributes list or ID until the
-// end of line.
-const CommandAttributesMode = [
-  AttributeListStart,
-  Identifier,
-  Space,
-  { ...Newline, POP_MODE: true },
-];
 
 // Generates a Lexer for the given language comment patterns.
 export function makeLexer(languageTokens: TokenType[]): Lexer {
   const modes = {
     RootMode: makeRootMode(languageTokens),
-    CommandAttributesMode,
+    // After a command start tag, there may be an attributes list or ID until the
+   // end of line or a block comment end.
+    CommandAttributesMode: [
+      AttributeListStart,
+      Identifier,
+      ...tokenCategoryFilter(languageTokens, [BlockCommentEnd]),
```

Comment:
This now supports arbitrary block comment ends within the attribute list instead of after? I'm not sure I understand the need for this change.

Reply:
The reason for the change was that when the lexer was in "CommandAttributesMode" it was unable to parse a ``BlockCommentEnd`` token, so it gave an "unrecognized character error". I agree though that isn't super clear as it is currently written. 

I changed this so that "BlockCommentEnd" also pops the CommandAttributesMode lexer. Had a tricky issue where if a BlockCommentEnd was not defined it was unable to pass some tests, so added a helper function to check if "BlockCommentEnd" was passed..